readthedocs/proxito/middleware.py

```diff
@@ -22,6 +22,59 @@
 log = structlog.get_logger(__name__)  # noqa
 
 
+def _get_domain_from_host(host):
+    """
+    Get the normalized domain from a hostname.
+
+    A hostname can include the port.
+    """
+    return host.lower().split(":")[0]
+
+
+def _unresolve_domain(domain):
+    """
+    Unresolve domain by extracting relevant information from it.
+
+    :param str domain: Domain to extract the information from.
+    :returns: A tuple with: the project slug, domain object, and if the domain
+     is from an external version.
+    """
+    public_domain = _get_domain_from_host(settings.PUBLIC_DOMAIN)
+    external_domain = _get_domain_from_host(settings.RTD_EXTERNAL_VERSION_DOMAIN)
+
+    subdomain, *rest_of_domain = domain.split(".", maxsplit=1)
+    rest_of_domain = rest_of_domain[0] if rest_of_domain else ""
```

Comment:
Is there a reason we aren't using something like urlparse here? This logic feels brittle. 

This should probably also be more explicitly named:

```suggestion
    subdomain, *rest_of_domain = domain.split(".", maxsplit=1)
    root_domain = rest_of_domain[0] if rest_of_domain else ""
```